middle_end/flambda2/to_cmm/to_cmm_shared.ml

```diff
@@ -213,10 +213,9 @@ let const_static cst =
            (tag_targetint (Targetint_31_63.to_targetint i))) ]
   | Naked_float f -> [cfloat (Numeric_types.Float_by_bit_pattern.to_float f)]
   | Naked_float32 f ->
-    (* Here we are relying on the data section being zero initialized to
-       maintain the invariant that statically-allocated float32 values are zero
-       padded. *)
-    [cfloat32 (Numeric_types.Float32_by_bit_pattern.to_float f)]
+    (* Statically-allocated float32 values are zero padded. We must explicitly
+       add the padding otherwise subsequent values will be misaligned. *)
+    [cfloat32 (Numeric_types.Float32_by_bit_pattern.to_float f); cint32 0l]
```

Comment:
I've added a note, but without committing to behaviour.  For one thing I don't think it's worth the time to think about, as we are very unlikely to run this code on a big-endian system; and in any case, I think it depends on what instruction is used to read the value.

Reply:
It's not like `int32`, where as long as the padding bits encode the sign properly we can load a 64-bit int directly to avoid sign-extension. Here we will always load exactly the first 4 bytes, so this code is correct independently of the endianness.